kafka/src/main/java/org/axonframework/extensions/kafka/eventhandling/producer/DefaultProducerFactory.java

```diff
@@ -114,16 +115,16 @@ protected DefaultProducerFactory(Builder<K, V> builder) {
             return createTransactionalProducer();
         }
 
-        if (this.nonTransactionalProducer == null) {
+        if (nonTransactionalProducer.get() == null) {
             synchronized (this) {
-                if (this.nonTransactionalProducer == null) {
+                if (nonTransactionalProducer.get() == null) {
```

Comment:
Why not change this to an `AtomicReference#compareAndSet` operation?

Reply:
Kept the 'old' code, now replaced with `updateAndGet` which seems even more fitting.